src/mbgl/map/map_context.cpp

```diff
@@ -112,7 +112,12 @@ void MapContext::setStyleURL(const std::string& url) {
         if (res.status == Response::Successful) {
             loadStyleJSON(res.data, base);
         } else {
-            Log::Error(Event::Setup, "loading style failed: %s", res.message.c_str());
+            if (res.message.substr(res.message.length() - 3) == "404" &&
```

Comment:
Maybe we could do this here:
https://github.com/mapbox/mapbox-gl-native/blob/master/platform/darwin/http_request_nsurl.mm#L254

So we don't need to add knowledge of the FileSource backend to the MapContext.